src/transformers/generation/utils.py

```diff
@@ -2043,7 +2043,7 @@ def _prepare_cache_for_generation(
         )
         if generation_config.cache_implementation is not None:
             if generation_config.cache_implementation in NEED_SETUP_CACHE_CLASSES_MAPPING:
-                if generation_config.cache_implementation == "static" and not self._supports_static_cache:
+                if generation_config.cache_implementation == "static" and not self._can_compile_fullgraph:
                     raise ValueError(
```

Comment:
(outside the scope of this PR: we should have some easy way of knowing which models are compatible with which caches, and throw an appropriate exception here)